sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/implementation/OkHttpFileRequestBody.java

```diff
@@ -22,17 +22,12 @@ public OkHttpFileRequestBody(FileContent content, long effectiveContentLength, M
 
     @Override
     public void writeTo(BufferedSink bufferedSink) throws IOException {
-        long count = effectiveContentLength;
-        if (count < 0) {
-            // OkHttp marks chunked encoding as -1.
-            // The content length is not specified so sending all remaining content.
-            count = Long.MAX_VALUE;
-        }
         // RequestBody.create(File) does not support position and length.
         // BufferedSink implements WritableByteChannel so we can leverage FileChannel as source.
         // FileChannel supports positional reads.
         try (FileChannel channel = FileChannel.open(content.getFile(), StandardOpenOption.READ)) {
-            channel.transferTo(0, count, bufferedSink);
+            // FileContent.getLength always returs non-null.
+            channel.transferTo(content.getPosition(), content.getLength(), bufferedSink);
```

Comment:
Does this assume the file was transferred completely with a single `transferTo()` call?

Reply:
It does, which is wrong. thanks for pointing this out.

Reply:
@samvaity fixed. Thank you for pointing this out.